### PR TITLE
TiKV can pause/resume grpc when necessary

### LINKED ISSUE #15086: TiKV provides APIs to restart grpc service
When a TiKV node becomes unhealthy  (such as slow IO, network partition from certain nodes),  we usually want to fail fast existing requests and do leader transfer. 
However, we may not want to kill a tikv server when it's unhealthy because restarting the tikv is too expensive and it may impact performance (due to warmup cache). 
So there should be a way to do a soft restart that will only start/stop grpc services

### PR DESCRIPTION
**Pull request by tonyxuqqi**
### What is changed and how it works?

Issue Number: Close #15086

What's Changed:

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Provides status APIs to start and stop tikv's grpc services
```

### CODE REVIEW

**Comment by tonyxuqqi** [comment]
cc @LykxSassinator

**Comment by LykxSassinator** [comment]
This pr can be dropped as the "https://github.com/tikv/tikv/pull/15088" has been ready for review and included all modifications in it.

**Comment by tonyxuqqi** [comment]
close it. We're using #15088

File: src/server/server.rs [hunk 1]
```diff
@@ -344,6 +394,35 @@ where
         Ok(())
     }
 
+    pub fn pause(&mut self) -> Result<()> {
+        info!("pausing the grpc server");
```
Comment (tonyxuqqi) [inline src/server/server.rs:398]: Taining's point is valid. It's better to add a time watch on this function. Same for the resume.

**Pull request closed without merging**